cpp/src/fil/fil.cu

```diff
@@ -586,20 +586,35 @@ void tl2fil_common(forest_params_t* params, const tl::Model& model,
   // fill in forest-dependent params
   params->depth = max_depth(model);  // also checks for cycles
 
+  const tl::ModelParam& param = model.param;
+
   // assuming either all leaves use the .leaf_vector() or all leaves use .leaf_value()
   size_t leaf_vec_size = tl_leaf_vector_size(model);
   if (leaf_vec_size > 0) {
     ASSERT(leaf_vec_size == model.num_output_group,
            "treelite model inconsistent");
     params->num_classes = leaf_vec_size;
     params->leaf_payload_type = leaf_value_t::INT_CLASS_LABEL;
+    ASSERT(
+      param.pred_transform != "identity",
```

Comment:
We should only allow the multi-class prediction transforms from here (https://github.com/dmlc/treelite/blob/mainline/src/compiler/pred_transform.cc#L67) that we support. At the moment, those transforms are `identity_multiclass` and `max_index`.